src/main/kotlin/com/openlattice/indexing/BackgroundLinkingIndexingService.kt

```diff
@@ -103,54 +105,117 @@ class BackgroundLinkingIndexingService(
     private val unIndexCandidates = hazelcastInstance
             .getQueue<Triple<List<Array<UUID>>, UUID, OffsetDateTime>>(HazelcastQueue.LINKING_UNINDEXING.name)
 
+    private val limiter = Semaphore(indexerConfiguration.parallelism)
+
     @Suppress("UNUSED")
-    private val linkingIndexingWorker = executor.submit {
-        generateSequence(indexCandidates::take)
-                .chunked(LINKING_INDEX_SIZE)
-                .asStream()
-                .parallel()
-                .forEach { candidateBatch ->
-                    if (!indexerConfiguration.backgroundLinkingIndexingEnabled) {
-                        return@forEach
-                    }
-
-                    val (linkingEntityKeyIdsWithLastWrite, linkingIds) = mapCandidates(candidateBatch)
-
-                    try {
-                        lock(linkingIds)
-                        index(linkingEntityKeyIdsWithLastWrite, linkingIds)
-                    } catch (ex: Exception) {
-                        logger.error("Unable to index linking entity with from batch of linking ids $linkingIds.", ex)
-                    } finally {
-                        unLock(linkingIds)
-                    }
+    @Scheduled(fixedRate = LINKING_INDEX_RATE)
+    private fun linkingIndexing() {
+        if (isLinkingIndexingEnabled()) {
```

Comment:
I'd prefer this as 
```
if (isLinkingIndexingEnabled()) {
  return
}

executor.submit {
  ...
}
```

just to reduce the amount of nesting here, since it gets pretty deep